registry/endpoint.go

```diff
@@ -227,6 +227,21 @@ func (e *Endpoint) pingV2() (RegistryInfo, error) {
 	}
 	defer resp.Body.Close()
 
+	// The endpoint may have multiple supported versions.
+	// Ensure it supports the v2 Registry API.
+	var supportsV2 bool
+
+	for _, versionName := range resp.Header[http.CanonicalHeaderKey("Docker-Distribution-API-Version")] {
```

Comment:
@jlhawn This might be too late: this loop is only iterating over the instances of the `Docker-Distribution-API-Version`. There needs to be a split on space for this to work correctly in the future.

For example, this only supports this:

```
Docker-Distribution-API-Version: registry/2.0
Docker-Distribution-API-Version: registry/2.1
```

When it should support this:

```
Docker-Distribution-API-Version: registry/2.0 registry/2.1
```

This will keep 1.5 sufficiently future proof.

And even this:

```
Docker-Distribution-API-Version: registry/2.0 registry/2.1
Docker-Distribution-API-Version: registry/1.0
```